src/OpenTelemetry/Trace/TracerProviderSdk.cs

```diff
@@ -372,6 +372,7 @@ private void RunGetRequestedDataAlwaysOnSampler(Activity activity)
         private void RunGetRequestedDataAlwaysOffSampler(Activity activity)
         {
             activity.IsAllDataRequested = false;
+            activity.ActivityTraceFlags &= ActivityTraceFlags.None;
```

Comment:
Do we need `&=` or `=` is sufficient?

Reply:
`=` should be sufficient as these flags are exclusive. I was just following the existing `|=` approach for updating the ActivityTraceFlags to `Recorded`.

Reply:
I think `|=` makes sense if we are turning a certain bit/flag to `1`. `&= None` doesn't make much sense since we're not clearing a specific bit/flag here.

Reply:
As of today, we only have two values for `ActivityTraceFlags`: `None: 0` and `Recorded: 1`. For now, both of these approaches `&=` and `=` will set the value of `ActivityTraceFlags` to `0`.

I think the correct thing to do would be to clear the specific bit as we might have some more values added to the enum later on. How about this: **`activity.ActivityTraceFlags &= ~ActivityTraceFlags.Recorded;`**? This should clear the bit set for `Recorded`.

Reply:
Sounds good to me. 👍

Reply:
Fixed this.